lib/streamlit/runtime/scriptrunner/script_runner.py

```diff
@@ -394,203 +394,209 @@ def _run_script(self, rerun_data: RerunData) -> None:
         """
         assert self._is_in_script_thread()
 
-        _LOGGER.debug("Running script %s", rerun_data)
-
-        start_time: float = timer()
-        prep_time: float = 0  # This will be overwritten once preparations are done.
-
-        # Reset DeltaGenerators, widgets, media files.
-        runtime.get_instance().media_file_mgr.clear_session_refs()
-
-        main_script_path = self._main_script_path
-        pages = source_util.get_pages(main_script_path)
-        # Safe because pages will at least contain the app's main page.
-        main_page_info = list(pages.values())[0]
-        current_page_info = None
-        uncaught_exception = None
-
-        if rerun_data.page_script_hash:
-            current_page_info = pages.get(rerun_data.page_script_hash, None)
-        elif not rerun_data.page_script_hash and rerun_data.page_name:
-            # If a user navigates directly to a non-main page of an app, we get
-            # the first script run request before the list of pages has been
-            # sent to the frontend. In this case, we choose the first script
-            # with a name matching the requested page name.
-            current_page_info = next(
-                filter(
-                    # There seems to be this weird bug with mypy where it
-                    # thinks that p can be None (which is impossible given the
-                    # types of pages), so we add `p and` at the beginning of
-                    # the predicate to circumvent this.
-                    lambda p: p and (p["page_name"] == rerun_data.page_name),
-                    pages.values(),
-                ),
-                None,
+        while True:
```

Comment:
nit: maybe a comment here would be nice why we need/use a `while True` loop here.